Tensile/TensileCreateLibrary.py

```diff
@@ -1059,6 +1061,69 @@ def verifyManifest(manifest) -> bool:
   
   return False if len(files) > 0  else True
 
+def parseGfxArchitectures(inputArchs: str) -> Set[str]:
+    """Parse target architectures from format **'arch1;arch2'** or **'arch1_arch2'** to **{'arch1', 'arch2'}**
+
+    Args:
+        inputArchs: Semicolon or underscore delimited Gfx architectures.
+
+    Returns:
+        Target Gfx architecture names.
+
+    Raises:
+        ValueError: If **inputArchs** (i) refers to an unknown Gfx architecture,
+            or (ii) uses incorrect delimiters.
+
+    """
+    inputArchs = inputArchs.lower()
+
+    # CLI uses `;` delimiters, CMake uses `_` delimiters
+    archs = inputArchs.split(";" if ";" in inputArchs else "_")
+
+    output = set()
+    for arch in archs:
+        if arch == "all":
+            return set([arch])
+        elif arch in output:
+            printWarning(f"Removing duplicate architecture {arch}.")
+            continue
+        elif arch == "":  # Omit empty archs caused by leading and trailing delimiters
+            continue
+        elif arch not in architectureMap:
+            raise ValueError(f"`{arch}` is not a known Gfx architecture.")
+        else:
+            output.add(arch)
+
+    if len(output) == 0:
+        raise ValueError(
+            "Could not parse a viable target architecture from {inputArchs}. When specifying multiple options, "
+            "use quoted, semicolon or underscore delimited Gfx architectures, e.g., `--architecture='gfx908;gfx1012'`"
+        )
+    return output
+
+
+def mapGfxArchitectures(gfxArchs: Set[str]) -> Set[str]:
+    """Validate and map target architectures from their Gfx name, to their common architecture name.
+
+    Args:
+        gfxArchs: Parsed Gfx architectures.
+
+    Returns:
+        Common architecture names associated with provided Gfx names. See **Common.architectureMap** for mappings.
+
+    Raises:
+        ValueError: If an unknown architecture is found.
+    """
+    if not isinstance(gfxArchs, set):
+        raise TypeError("`gfxArch` must be of type builtins.set")
+
+    namedArchs = set()
+    for arch in gfxArchs:
+        if arch in architectureMap:
+            namedArchs.add(architectureMap[arch])
+        else:
+            raise ValueError(f"`{arch}` is not a known Gfx architecture.")
```

Comment:
We do this validation in the other function - should we do this in both fxns?

Reply:
How do we know these functions will be called together in the future?